pkg/shares/reserved_bytes.go

```diff
@@ -0,0 +1,42 @@
+package shares
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+)
+
+// NewReservedBytes returns a byte slice of length
+// appconsts.CompactShareReservedBytes that contains a varint of the byteIndex
+// of the first unit that starts in a compact share. If no unit starts in the
+// compact share, ReservedBytes is [0, 0].
+func NewReservedBytes(byteIndex uint64) (reservedBytes []byte, err error) {
```

Comment:
One does not need to use naked returns when using named return values. Additionally a named return value is by default initialized to its zero value so these lines are unnecessary if they are specified as named return values:

```golang
        reservedBytes := []byte{}
        var err error
```

Agreed this is a cosmetic thing and I think @evan-forbes has similar opinions against named return values so I think we should have an offline discussion or issue to discuss it. My opinion is that:
1. named return values are good because they provide additional documentation in the function signature which is especially helpful when a function returns multiple values. They also enable the author to remove trivial initialization statements (like the two lines above).
2. naked returns are bad because they force the reader to return to the function signature to learn what values are being returned

So I would prefer if we used named return values but avoided naked returns.

Ref:
- https://yourbasic.org/golang/named-return-values-parameters/
- https://go.dev/tour/basics/7

Reply:
On second thought, we've already had a discussion on this: https://github.com/celestiaorg/celestia-core/issues/831 @SweeXordious feel free to comment or re-open that issue

Reply:
Aaah, I see. So, the decision is to leave it to developer preference? 
I personally have nothing against any.

Also, considering the initialization in the beginning, then this function can be written as:

```golang
func NewReservedBytes(byteIndex uint64) (reservedBytes []byte, err error) {
	if byteIndex >= appconsts.ShareSize {
		err = fmt.Errorf("byte index %d must be less than share size %d", byteIndex, appconsts.ShareSize)
                return
	}
	reservedBytes = make([]byte, appconsts.CompactShareReservedBytes)
	bytesWritten := binary.PutUvarint(reservedBytes, byteIndex)
	if bytesWritten > appconsts.CompactShareReservedBytes {
		err = fmt.Errorf("byte index %d must occupy less than %d bytes when written as varint", byteIndex, appconsts.CompactShareReservedBytes)
                return
	}
	return
}
```

My point is, if we're using named returns, then we should use them also inside the function, such as:

```
err = fmt.Errorf("byte index %d must occupy less than %d bytes when written as varint", byteIndex, appconsts.CompactShareReservedBytes)
```
instead of:
```
return []byte{}, fmt.Errorf("byte index %d must occupy less than %d bytes when written as varint", byteIndex, appconsts.CompactShareReservedBytes)
```

But I don't have a strong opinion really.

Reply:
I think explicit returns are more readable than naked returns so my preference is:

```golang
return []byte{}, fmt.Errorf("byte index %d must occupy less than %d bytes when written as varint", byteIndex, appconsts.CompactShareReservedBytes)
```
 instead of:

```golang
err = fmt.Errorf("byte index %d must be less than share size %d", byteIndex, appconsts.ShareSize)
return
```

but I'll remove usages of named return values from this PR to resolve this

Reply:
Resolved in https://github.com/celestiaorg/celestia-app/pull/850/commits/8a6a66bb81294774b915b9763983e75a6559f76c

Reply:
Btw, I'm okey with both, I'm just trying to see which one is better. It's alright to keep any from my perspective.
Btw, as a general rule, when I'm implementing stuff, should I use the naked or explicit returns? or it's left as a developer preference?
Thanks a lot for  your help and time :pray:

Reply:
No worries. Unfortunately it doesn't seem like there's an opinionated Go convention on it so it's left to developer presence. 

Since this codebase largely doesn't use names return params, I think it makes sense to conform to that.

Reply:
Awesome, thanks a lot again :pray: